### PR TITLE
Fix BRP069: Add swing_mode support for Alira X devices

### PR DESCRIPTION
**Pull request by kingy444**
I am trying to fix some broken tests and found these missing.

Seem to be required for `tests/test_daikin_alira_x.py` and were removed at somepoint

I dont have one of these to test with, but assume the tests were written with these in mind

### CODE REVIEW

**Comment by fredrike** [comment]
Thank you for finding this!

This is the missing code: https://github.com/fredrike/pydaikin/compare/v2.11.1...v2.13.4#diff-92a2c94a05a479c70175c80e0a123bddeb645754f88ac774cb7b43b55dd5db10L180-L186

I think we should try to re-add it to the same place, after this: https://github.com/fredrike/pydaikin/blob/2debda532307448fc6b464f297167b8ab918de1b/pydaikin/daikin_brp069.py#L181 but as `param` instead.

The code could look like this:

```python
            if 'f_dir_lr' in self.values and 'f_dir_ud' in self.values:
                # Australian Alira X uses 2 separate parameters instead of the combined f_dir
                params["f_dir_ud"] = 'S' if self.values['f_dir'] in ('1', '3') else '0'
                params["f_dir_lr"] = 'S' if self.values['f_dir'] in ('2', '3') else '0'
            else:
                params["f_dir"] = self.values['f_dir']
```

I think this is nice but perhaps not clearer:

```python
if all(i in self.values for i in ('f_dir_lr', 'f_dir_ud')):
```

File: pydaikin/response.py [hunk 1]
```diff
@@ -15,4 +15,15 @@ def parse_response(response_body):
...
+
+    # Translate swing mode from 2 parameters to 1 (Special case for certain models e.g Alira X)
+    if response.get("f_dir_ud") == "0" and response.get("f_dir_lr") == "0":
+        response["f_dir"] = '0'
+    if response.get("f_dir_ud") == "S" and response.get("f_dir_lr") == "0":
+        response["f_dir"] = '1'
+    if response.get("f_dir_ud") == "0" and response.get("f_dir_lr") == "S":
+        response["f_dir"] = '2'
+    if response.get("f_dir_ud") == "S" and response.get("f_dir_lr") == "S":
+        response["f_dir"] = '3'
```
Comment (fredrike) [inline pydaikin/response.py:27]:
```suggestion
    f_dir = 0
    if response.get("f_dir_ud") == "S":
        f_dir += 1
    if response.get("f_dir_lr") == "S":
        f_dir += 2
    if "f_dir" not in response:
        response["f_dir"] = f_dir
```

Comment (fredrike) [inline pydaikin/response.py:27]: shorter but perhaps harder to read..

```suggestion
    if "f_dir" not in response:
        response["f_dir"] = (1 if response.get("f_dir_ud") == "S" else 0) + (
            2 if response.get("f_dir_lr") == "S" else 0
        )
```

Comment (fredrike) [reply pydaikin/response.py:27]: The idea is that `S` seems to add 1 or 2 to `f_dir`

**Comment by fredrike** [comment]
@kel30a  should also be included in this discussion.

**Comment by kel30a** [comment]
Tis a shame the pytest tests weren't previously running in the github workflows. I noticed @fredrike just added that in on master. Does @kingy444 need to bring those changes into his fork to make the test run as a "check" on this PR?

File: pydaikin/response.py (hunk 1 above)
Comment (kel30a) [reply pydaikin/response.py:27]: I think I prefer the more older, more explicit form

File: pydaikin/response.py [hunk 2]
```diff
@@ -15,4 +15,15 @@ def parse_response(response_body):
         return {}
     if 'name' in response:
         response['name'] = unquote(response['name'])
+
```
[the remark this answers is not in the document]
Comment (kel30a) [reply pydaikin/response.py:18]: Agreed

**Review by kel30a — CHANGES_REQUESTED** [review]

**Comment by fredrike** [comment]
> Tis a shame the pytest tests weren't previously running in the github workflows. I noticed @fredrike just added that in on master. Does @kingy444 need to bring those changes into his fork to make the test run as a "check" on this PR?
> 
> Happy to test once suggested changes have been made.

Well, BitBucket broke for us so I had to move the repository here and some info was lost in the transition.

The tests have not really been enforced but once we are back on track I'll enable them. I think a resync with master is a good thing here.

```
git fetch --all
git merge origin/master
```

File: pydaikin/response.py (hunk 1 above)
Comment (fredrike) [reply pydaikin/response.py:27]: I agree but at least it should be `elif`
```suggestion
    if response.get("f_dir_ud") == "0" and response.get("f_dir_lr") == "0":
        response["f_dir"] = '0'
    elif response.get("f_dir_ud") == "S" and response.get("f_dir_lr") == "0":
        response["f_dir"] = '1'
    elif response.get("f_dir_ud") == "0" and response.get("f_dir_lr") == "S":
        response["f_dir"] = '2'
    elif response.get("f_dir_ud") == "S" and response.get("f_dir_lr") == "S":
        response["f_dir"] = '3'
```

**Comment by kingy444** [comment]
Enforcing tests might cause an issue until they are all sorted right?

There are other failing tests too that I haven't gotten around to fixing. 

I did start but it's fun not having these devices. 

I found the power tests are now asserting false where previously they were true... 

Then even if I fake that assertion result the next bunch of checks fail too. Again, assuming that everything worked once upon a time I don't know whether the test is bad or the library 😂

**Comment by fredrike** [comment]
Well,having tests would probably have identified some of the errors we have introduced lately. But building good tests are hard too. I'm not sure what I think of this.

But great if we can sort the altaria issues for now.

**Comment by kingy444** [comment]
> @kingy444 do you have time to fix this in the next couple of hours so I can release a new pydaikin version with the SkyFi fix together with this?

trying to pull something together now for you

File: pydaikin/daikin_brp069.py [hunk 3]
```diff
@@ -117,6 +117,26 @@ class DaikinBRP069(Appliance):
...
     }
 
+    @staticmethod
+    def parse_response(response_body):
+        """Parse response from Daikin
+
+        Translate swing mode from 2 parameters to 1 (Special case for certain models e.g Alira X)
+        """
+        _LOGGER.debug("Parsing %s", response_body)
+        response = super(DaikinBRP069, DaikinBRP069).parse_response(response_body)
```
Comment (fredrike) [inline pydaikin/daikin_brp069.py:127]: I'm not super good at working with supclasses and `super` but the [documentation](https://docs.python.org/3/library/functions.html#super) suggests this:

> If the second argument is omitted, the super object returned is unbound. If the second argument is an object, isinstance(obj, type) must be true. If the second argument is a type, issubclass(type2, type) must be true (this is useful for classmethods).

```python
class C(B):
    def method(self, arg):
        super().method(arg)    # This does the same thing as: 
                                               # super(C, self).method(arg)
```

```suggestion
        response = super().parse_response(response_body)
```

**Review by fredrike — CHANGES_REQUESTED** [review]

File: pydaikin/daikin_brp069.py (hunk 3 above)
Comment (fredrike) [reply pydaikin/daikin_brp069.py:127]: Oh, `parse_response` is a static method so I guess it is ok to have your code..

**Pull request merged**